libvirt/cloudinit_def.go

```diff
@@ -13,33 +13,23 @@ import (
 	"time"
 
 	"github.com/hooklift/iso9660"
-	"github.com/imdario/mergo"
 	libvirt "github.com/libvirt/libvirt-go"
 	"github.com/mitchellh/packer/common/uuid"
 	"gopkg.in/yaml.v2"
 )
 
-// userData is the filename expected by cloud-init
-const userData string = "user-data"
-
 // metaData is the filename expected by cloud-init
 const metaData string = "meta-data"
 
-type defCloudInitUserData struct {
-	SSHAuthorizedKeys []string `yaml:"ssh_authorized_keys"`
-}
-
 type defCloudInitMetaData struct {
-	LocalHostname string `yaml:"local-hostname,omitempty"`
-	InstanceID    string `yaml:"instance-id"`
+	InstanceID string `yaml:"instance-id"`
```

Comment:
need to check this more

Reply:
I would say these are implementation details of what cloud-init support or not.

I would favor the approach where user can set whatever they want here, hopefully something cloud-init support (adding links towards the relevant cloud-init documentation would be great I guess).
Having default values like today for the Instance ID is nice (although I guess the instance ID itself should be changed, I have something currently named like `created-at-2018-09-05 18:53:16.5935747 +0200 CEST m=+37.022498553` :D)

Also, might be good to have support [for the other cloud-init fields](https://git.launchpad.net/cloud-init/tree/cloudinit/sources/DataSourceNoCloud.py#n71)?

Reply:
Actually, I think you need to let the metadata field, but:

* make it a string so that user can populate it as they want + document that if it's being filled in by users, they have to conform to what cloud-init needs (apparently, only `instance-id` is required)
* if it's empty, generate a `instance-id` field like it is done today

Since `meta-data` will not be filled in automatically by libvirt, if a cloud-init datasource is provided by the Terraform provider, then I would say it's up to the user to set it correctly! :)